express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+var express = require('express');
+var app = express();
+
+const data = require('./data.json');
+
+app.get('/api/notes', function (req, res) {
+  var notesArray = [];
+  for (const property in data.notes) {
+    notesArray.push(data.notes[property]);
+  };
+  res.status(200);
+  res.json(notesArray);
+});
+
+app.use(express.json());
+
+app.get('/api/notes/:id', function (req, res) {
+  var notesId = req.params.id;
+  if (Math.sign(notesId) === 1) {
```

Comment:
While this catches non-positive `id`s, it doesn't catch positive fractional numbers, which get the wrong error message.